Classes/EventListener/EventListenerInvoker.php

```diff
@@ -31,111 +30,153 @@ final class EventListenerInvoker
     private $eventStore;
 
     /**
-     * @Flow\Inject
-     * @var EntityManagerInterface
+     * @var EventListenerInterface
      */
-    protected $entityManager;
+    private $eventListener;
 
-    public function __construct(EventStore $eventStore)
+    /**
+     * DBAL connection for the DoctrineAppliedEventsStorage (@see getAppliedEventsStorageForListener())
+     *
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var \Closure[]
+     */
+    private $progressCallbacks = [];
+
+    /**
+     * How many events should be applied until the whole transaction is committed
+     * and the highest applied sequence number is persisted.
+     * By default the transaction is committed for every event (batch size = 1)
+     *
+     * @var int
+     */
+    private $transactionBatchSize = 1;
+
+    public function __construct(EventStore $eventStore, EventListenerInterface $eventListener, Connection $connection)
```

Comment:
I dislike that we have to pass in the DBAL connection here as it seems unrelated.. But it seems like an OKish compromise for now that can be replaced without major breaking changes hopefully